src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1801 @@
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Loaded StatusLightBrightness as %d", to_underlying(mStatusLightBrightness));
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Unable to load the StatusLightBrightness from the KVS. Defaulting to %d",
+                      to_underlying(mStatusLightBrightness));
+    }
+
+    // Signal delegate that all persistent configuration attributes have been loaded.
+    mDelegate.PersistentAttributesLoadedCallback();
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::StoreViewport(const ViewportStruct & viewport)
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(viewport.Encode(writer, TLV::AnonymousTag()));
+
+    return mPersistentStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::CameraAVStreamMgmtViewport().KeyName(), buffer,
+                                               static_cast<uint16_t>(writer.GetLengthWritten()));
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::ClearViewport()
+{
+    // TODO
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadViewport(ViewportStruct & viewport)
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    MutableByteSpan bufferSpan(buffer);
+    uint16_t size = static_cast<uint16_t>(bufferSpan.size());
+
+    ReturnErrorOnFailure(mPersistentStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::CameraAVStreamMgmtViewport().KeyName(),
+                                                             bufferSpan.data(), size));
+    bufferSpan.reduce_size(size);
+
+    TLV::TLVReader reader;
+
+    reader.Init(bufferSpan);
+    ReturnErrorOnFailure(reader.Next(TLV::AnonymousTag()));
+    ReturnErrorOnFailure(viewport.Decode(reader));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::StoreRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+    TLV::TLVWriter writer;
+    writer.Init(buffer);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Array, arrayType));
+
+    for (uint8_t i = 0; i < kNumOfStreamUsageTypes; i++)
+    {
+        ReturnErrorOnFailure(writer.Put(TLV::AnonymousTag(), mRankedVideoStreamPriorities[i]));
+    }
+    ReturnErrorOnFailure(writer.EndContainer(arrayType));
+
+    return mPersistentStorage->SyncSetKeyValue(
+        DefaultStorageKeyAllocator::CameraAVStreamMgmtRankedVideoStreamPriorities().KeyName(), buffer,
+        static_cast<uint16_t>(writer.GetLengthWritten()));
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::LoadRankedVideoStreamPriorities()
+{
+    uint8_t buffer[kRankedVideoStreamPrioritiesTlvSize];
+
+    uint16_t len = kRankedVideoStreamPrioritiesTlvSize;
+    ReturnErrorOnFailure(mPersistentStorage->SyncGetKeyValue(
+        DefaultStorageKeyAllocator::CameraAVStreamMgmtRankedVideoStreamPriorities().KeyName(), buffer, len));
+
+    TLV::TLVReader reader;
+    reader.Init(buffer);
```

Comment:
Shouldn't this somehow take `len` into account?  The viewport codepath did this right.